coalib/settings/Section.py

```diff
@@ -159,6 +160,10 @@ def add_or_create_setting(self,
             val.value = str(val._value) + '\n' + setting._value
             self.append(val, custom_key=key)
         else:
+            if key in self.contents:
+                logging.warning('{} setting has already been defined in '
+                                'section {}. The previous setting will be '
+                                'overridden.'.format(key, self.name))
```

Comment:
Actually the function does what it says, I'm not sure we should alter that behaviour. Imo it belongs to `ConfParser`. Any reason not to do it inside `ConfParser`?